profile/src/main/java/rdx/works/profile/domain/backup/RestoreAndSkipMainSeedPhraseUseCase.kt

```diff
@@ -0,0 +1,43 @@
+package rdx.works.profile.domain.backup
+
+import rdx.works.profile.data.model.apppreferences.Radix
+import rdx.works.profile.data.model.extensions.addMainBabylonDeviceFactorSource
+import rdx.works.profile.data.model.extensions.changeGateway
+import rdx.works.profile.data.model.factorsources.DeviceFactorSource
+import rdx.works.profile.data.repository.BackupProfileRepository
+import rdx.works.profile.data.repository.DeviceInfoRepository
+import rdx.works.profile.data.repository.MnemonicRepository
+import rdx.works.profile.data.repository.ProfileRepository
+import java.time.Instant
+import javax.inject.Inject
+
+class RestoreAndSkipMainSeedPhraseUseCase @Inject constructor(
```

Comment:
I am not sure why we need this extra use case when we already have the `EnsureBabylonFactorSourceExistUseCase` which does the same thing?

The restore is always on the mainnet right? So what's the point of not using the `EnsureBabylonFactorSourceExistUseCase`?